databricks/koalas/indexes.py

```diff
@@ -419,6 +419,34 @@ def fillna(self, value):
         result = DataFrame(self._kdf._internal.copy(sdf=sdf)).index
         return result
 
+    # TODO: ADD keep parameter
+    def drop_duplicates(self):
+        """
+        Return Index with duplicate values removed.
+
+        Returns
+        -------
+        deduplicated : Index
+
+        See Also
+        --------
+        Series.drop_duplicates : Equivalent method on Series.
+        DataFrame.drop_duplicates : Equivalent method on DataFrame.
+
+        Examples
+        --------
+        Generate an pandas.Index with duplicate values.
+
+        >>> idx = ks.Index(['lama', 'cow', 'lama', 'beetle', 'lama', 'hippo'])
+
+        >>> idx.drop_duplicates() # doctest: +SKIP
+        Index(['lama', 'cow', 'beetle', 'hippo'], dtype='object')
+        """
+        sdf = self._internal.sdf.select(self._internal.index_scols).drop_duplicates()
+        internal = _InternalFrame(sdf=sdf, index_map=self._internal.index_map)
```

Comment:
`index_map=self._kdf._internal.index_map` rather than `index_map=self._internal.index_map` reason of https://github.com/databricks/koalas/pull/1120#discussion_r357491784